common/include/pcl/PCLHeader.h

```diff
@@ -44,8 +42,9 @@ namespace pcl
 
   inline bool operator== (const PCLHeader &lhs, const PCLHeader &rhs)
   {
-    return (&lhs == &rhs) ||
-      (lhs.seq == rhs.seq && lhs.stamp == rhs.stamp && lhs.frame_id == rhs.frame_id);
+    return lhs.seq == rhs.seq &&
```

Comment:
Why is the check on pointer location removed? If the 2 pointers are same, this will short circuit the comparison. Your version forces a complete check

Reply:
will restore it